src/Vector.js

```diff
@@ -16,63 +16,58 @@ class Vector
 	 */
 	static Create(str)
 	{
-		const allVectors = [];
-		const substrings = str.split(";");
+		const vectors = [];
 		
-		/* Helper function */
-		function CrossProduct(arr1, arr2)
-		{
-			const crossProduct = [];
-			for (let i = 0; i < arr1.length; i++)
-			{
-				for (let j = 0; j < arr2.length; j++)
-				{
-					crossProduct.push([arr1[i], arr2[j]]);
-				}
-			}
-			return crossProduct;
-		}
-		
-		substrings.forEach((substring) => {
+		// Iterate over each vector string, delimited by semicolon
+		str.split(";").forEach((substring) => {
 			const vectorString = substring.trim();
-			if (vectorString == "")
+			if (vectorString === "")
 			{
 				/* Ignore empty strings - common if vector list ended with semicolon */
 				return;
 			}
 			
-			const checkValid = vectorString.match(/\(-?\d+[\d{}+jhpmd]*, -?\d+[\d{}+jhpmd]*\)[\d{}+jhpmd]*/g);
-			if (checkValid == undefined || checkValid.length <= 0)
+			const checkValid = vectorString.match(/\ *\(\ *-?\d+[\d{}+jhpmd]*\ *,\ *-?\d+[\d{}+jhpmd]*\ *\)[\d{}+jhpmd]*\ */g);
```

Comment:
Don't need to accept whitespace at either end - that's why we trim the string.

Reply:
I actually assumed no whitespace, except after the comma in the middle. Would suggest instead using this. It allows any number of spaces after the comma, but no other spaces.
```suggestion
			const checkValid = vectorString.match(/\(-?\d+[\d{}+jhpmd]*,\ *-?\d+[\d{}+jhpmd]*\)[\d{}+jhpmd]*/g);
```

Reply:
I have two justifications for allowing whitespace pretty much everywhere: one stronger than the other.

The weaker argument: I don't want people to have to hunt down an accidental fat-thumb-extra-space in their notation. Seems like a condition that we don't need them to worry about when they're trying to make a game.

The stronger argument: I think tabulating notation is rather convenient. Especially as a vector notation grows more complex for weird pieces, users might be interested in tabulating their vectors like so:
```
(   1d, 2hp);
(10jpd,   1)
```
I think this is a reasonable use case that we should support.

PS: Probably weaker argument: Generally people have come to assume that parsers tend to ignore whitespace. I think we should facilitate this assumption.

Reply:
Fair enough. Sounds good.